### DISCUSSION TITLE
[Idea] "force update of friendly URL" setting should apply to categories and not just products

### DISCUSSION THREAD

**Discussion by jf-viguier** [PrestaShop/PrestaShop#32253] [Ideas]
#### Is your feature request related to a problem?

Category Friendly URL should not be updated if Force update of friendly URL is set to false in Product settings
There is no way to don't force the update of category friendly URL.

![image](https://user-images.githubusercontent.com/16067358/95731919-61bc2700-0c80-11eb-9963-a5ccd960b28d.png)

#### Describe the solution you'd like - edited by @LouiseBonnard on October 13th:

Instead of just adding the _Force update of friendly URL_ option to the categories page (as originally asked by @jf-viguier), we should turn it into a general SEO & URLs parameter that would apply to all URLs. Below is a screenshot example of the expected Shop Parameters > SEO & URLs page:

<img width="1548" alt="Capture d’écran 2020-10-13 à 15 49 46" src="https://user-images.githubusercontent.com/32565890/95869585-d3bd6a80-0d6b-11eb-802c-04b6fc5027ea.png">

#### Steps to Reproduce

1. Go to Order settings and disable "Force update of friendly URL"
2. Go to a category page: change the category name and see that the friendly URL is changed.

* PrestaShop version: 1.7.6
* PHP version: 7.2

[a comment by an automated account is not shown]

**Comment by khouloudbelguith** [comment]
Hi @jf-viguier,

The option "Force update of friendly URL" is added in the Shop Parameters > Product Settings.
![image](https://user-images.githubusercontent.com/16067358/95721145-75609100-0c72-11eb-90dc-89980d8d4d53.png)
It is applied on Saving Products, not related to the category page.
In the Category page, the Friendly URL should automatically be updated as reported here: https://github.com/PrestaShop/PrestaShop/issues/20931

Thanks to check and feedback.

**Comment by jf-viguier** [comment]
@khouloudbelguith the issue is about creation time. My concern is update time.
There is no way to don't force update of category friendly url.
I know that this configuration page is about product but there is no category configuration page.
I guess that if a merchand doesn't want to update friendly url on product page, it should be the same on category page.
By the way, google doesn't like so much 301 or 302 redirections and friendly url change launch a redirection so it's better to keep url constant over time.

**Comment by khouloudbelguith** [comment]
@jf-viguier,

Thanks for these clarifications.
So, it is a new improvement that could be added.
Ping @PrestaShop/product-team what do you think about this?
Can we add this option "Force update of friendly URL" to the categories page?
 
Epic related: https://github.com/PrestaShop/PrestaShop/issues/18402

**Comment by clotairer** [comment]
It's a bug for now, isn't it ?
The force url generation in products should be for all catalog objects (products and categories). Or by default URL should not be modified automatically. If not, we will have a SEO issue.

**Comment by jf-viguier** [comment]
I agree with you @clotaire202, thanks for the support. @PrestaShop/product-team WDYT ?

**Comment by LouiseBonnard** [comment]
It makes more sense to have this option available for products and categories. It should be available in Shop Parameters > General then. What about attributes, features, brands, and suppliers... or simply turning it into a general parameter? Do you need it for all URLs or mainly the products and categories URLs?

**Comment by jf-viguier** [comment]
Good idea @LouiseBonnard, it should be a general parameter and impact all prestashop friendly urls.

**Comment by LouiseBonnard** [comment]
I found it more relevant to add it as a setting in the SEO & URLs page, see the specifications above. If it is okay for you, I will update this issue description accordingly. :-)

**Comment by jf-viguier** [comment]
@LouiseBonnard ok for me. It's better in SEO & URLs page

**Comment by valentin-anamorphik** [comment]
@jf-viguier is it possible don't force the update of category friendly URL with a module or override?

This option should be impact all prestashop friendly urls.
